src/main.rs

```diff
@@ -1093,23 +1200,87 @@ fn find_device() -> Result<PathBuf> {
     Ok(path)
 }
 
-fn hid_recorder() -> Result<()> {
-    let cli = Cli::parse();
+fn event_handler(data: &[u8], start_time: &mut Option<Instant>) -> ::std::os::raw::c_int {
+    if data.len() != std::mem::size_of::<hid_recorder_event>() {
+        eprintln!(
+            "Invalid size {} != {}",
+            data.len(),
+            std::mem::size_of::<hid_recorder_event>()
+        );
+        return 1;
+    }
 
-    let mut stream: Box<dyn Write> = if cli.output_file == "-" {
-        // Bit lame but easier to just set the env for owo_colors to figure out the rest
-        match cli.color {
-            ColorChoice::Never => std::env::set_var("NO_COLOR", "1"),
-            ColorChoice::Auto => {}
-            ColorChoice::Always => std::env::set_var("FORCE_COLOR", "1"),
-        }
+    let event = unsafe { &*(data.as_ptr() as *const hid_recorder_event) };
 
-        Box::new(std::io::stdout())
+    *start_time = if start_time.is_none() {
+        Some(Instant::now())
     } else {
-        std::env::set_var("NO_COLOR", "1");
-        Box::new(std::fs::File::create(cli.output_file).unwrap())
+        *start_time
     };
 
+    let elapsed = start_time.unwrap().elapsed();
+
+    if event.length == 0 {
+        return 1;
+    }
```

Comment:
```suggestion
    if event.length == 0 {
        return 1;
    }
    
    let elapsed = start_time.unwrap().elapsed();
```

we don't need elapsed if event.length is 0.


Arguably - do we want to set `*start_time` if `event.length == 0`? seems like a niche case and it's cleaner to do the various event checks first before getting into the time/elapsed/etc. handling